apps/gallery/test/marionette/lib/fakeactivitycaller.js

```diff
@@ -0,0 +1,83 @@
+/* global module */
+(function(module) {
+  'use strict';
+
+  var SELECTORS = Object.freeze({
+    pickMenu: 'form[data-z-index-level="action-menu"]',
+    pickImageButton: '#pick-image',
+    pickedImage: '#pick-activity-data > img',
+    pickedImageName: '#pick-activity-data > span.name',
+    pickedImageType: '#pick-activity-data > span.type',
+    sharedImage: '#share-activity-data > img',
+    sharedImageName: '#share-activity-data > span.name',
+    sharedImageType: '#share-activity-data > span.type'
+  });
+
+  var ORIGIN = 'gallery.fakeactivity.gaiamobile.org';
+
+  module.exports = {
+    ORIGIN: ORIGIN,
+
+    create: function(client) {
+      var originURL = 'app://' + ORIGIN;
+
+      return {
+        get pickMenu() {
+          // Switch to the system app first.
+          client.switchToFrame();
+          return client.helper.waitForElement(SELECTORS.pickMenu);
+        },
+
+        get pickedImage() {
+          return client.helper.waitForElement(SELECTORS.pickedImage);
+        },
+
+        get pickedImageName() {
+          return client.helper.waitForElement(SELECTORS.pickedImageName);
+        },
+
+        get pickedImageType() {
+          return client.helper.waitForElement(SELECTORS.pickedImageType);
+        },
+
+        get sharedImageName() {
+          return client.helper.waitForElement(SELECTORS.sharedImageName);
+        },
+
+        get sharedImageType() {
+          return client.helper.waitForElement(SELECTORS.sharedImageType);
+        },
+
```

Comment:
It would be cool to have an integration test helper class that would take an object like your SELECTORS object and automatically generate getters like these for you.  Not suggesting you spend the time to do that right now, unless that kind of meta-programming seems cool to you. Just saying that it would be useful here.